### PR TITLE
[4.0] Namespacing Finder Indexer classes

### PR DESCRIPTION
**Pull request by Hackwar**
This namespaces the Indexer classes of Finder/Smart Search. Besides namespacing, this fixes a few issues where variables were not correctly defined. For B/C, I've added empty files for the old classes, since quite a few finder plugins out there require_once these files.

### CODE REVIEW

**Comment by Hackwar** [comment]
I'm open for suggestions for the language classnames.

**Comment by Hackwar** [comment]
I've renamed them to first letter uppercase now.

File: administrator/components/com_finder/Indexer/Parser.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Finder\Administrator\Indexer;
+
+use Joomla\CMS\Filter\InputFilter;
+use Joomla\CMS\Language\Text;
+
+defined('_JEXEC') or die;
```
Comment (Quy) [inline administrator/components/com_finder/Indexer/Parser.php]: Move `defined` above `use` statements.

**Comment by chmst** [comment]
I have tested this with some use cases and it looks good to me. Everything was as expected. 
But to say the truth, I did not inspect every line of code and nobody can test all possible constellations. 
Maybe it is the best to merge this and then fix what might follow.

**Comment by Hackwar** [comment]
@wilsonge asked me to commit this again to prevent future merge conflicts. I moved the files first and then did changes to the content. Hopefully that should allow git to understand the changes better. The changes are the same as before.

**Comment by wilsonge** [comment]
Thanks for recommitting!

**Comment by Hackwar** [comment]
Thank you for merging. 😉

**Pull request merged**